### ISSUE TITLE
Custom properties theme 

### ISSUE THREAD

**Issue by Bartvelp** [molstar/molstar#404]
Hi,

I'm looking to use Molstar in my project and would like to color parts of the protein differently. Very similar to #342, in that issue
@dsehnal said this:
> Hi,
> 
> yes, you can define custom themes to which you can pass properties to. Here's an example: https://github.com/molstar/molstar/blob/master/src/examples/basic-wrapper/index.ts#L108. You can pass the props to the theme say as `{ color: 'theme-name', colorParams: { ranges: [[1, 10], [15, 25]] }}`. [Here's](https://github.com/molstar/molstar/blob/master/src/extensions/pdbe/structure-quality-report/color.ts#L44) an example of a more complex theme which allows you pass props.
> 
> Let us know if you need more help.

But I still can not figure out what I need to add to a basic theme ([like this one](https://github.com/molstar/molstar/blob/master/src/examples/basic-wrapper/coloring.ts)) to actually "receive" the props in the theme, I don't really understand the flow of defining "custom properties" and then passing them to the theme. To me this sounds like a basic functionality so I don't really understand why it is this difficult. 

I also tried to augment the [ResidueNameColorTheme](https://github.com/molstar/molstar/blob/2458ea7b9251965309af403ba4475a8c02843693/src/mol-theme/color/residue-name.ts), because I expected the props I defined in `colorParams` to be passed down to `props` here:
https://github.com/molstar/molstar/blob/2458ea7b9251965309af403ba4475a8c02843693/src/mol-theme/color/residue-name.ts#L100
But this unfortunately this is not the case, 
Overall I would need to congratulate the devs on writing such a strongly typed project, really amazing! Would be nice if there were more docs available though!

**Comment by dsehnal** [comment]
Hi,

have a look at https://github.com/molstar/molstar/blob/master/src/extensions/pdbe/structure-quality-report/color.ts and the surrounding files. That might be what you need.

> Would be nice if there were more docs available though!

Agree. If you want to help, we accept PRs at https://github.com/molstar/docs.

**Comment by Bartvelp** [comment]
Thanks again for your response, it helped me track down where I made my mistake.
I assumed that the params in `colorParams` were passed down to props here:
https://github.com/molstar/molstar/blob/2458ea7b9251965309af403ba4475a8c02843693/src/mol-theme/color/residue-name.ts#L100

But I quickly dismissed it as I could not find them initially. I don't remember what I tried, but there params ***are*** passed down to props. Using `colorParams: { saturation: -6 }` for example I could set the saturation. I do not know why I did not try this before. I think I can figure it out from here.

Thanks for your help!
